File: Proc-0/Machine/Proc/HelpSymbol.cs
using System.Runtime.CompilerServices;

/// <summary>
/// Лексический анализатор документации.
/// </summary>
class HelpSymbol : Symbol
{
	/// <summary>
	/// Основной конструктор.
	/// </summary>
	/// <param name="charToken">Символьный поток.</param>
	public HelpSymbol(CharStream charToken) : base(charToken)
	{
	}

	[MethodImpl(MethodImplOptions.AggressiveInlining)]
	private void ReadDelimiter()
	{
	}

	[MethodImpl(MethodImplOptions.AggressiveInlining)]
	protected void ReadNumber()
	{
	}

	/// <summary>
	/// Чтение очередной лексемы. Простейший шаг, определяющий следующий автомат.
	/// Результат чтения не имеет значения, т.к. синтаксический анализатор сам решает,
	/// является ли для него состояние машины приемлемым для следующего шага.
	/// </summary>
	public override void Read()
	{
		//// Если нет возможности прочесть очередное состояние из потока, работа анализатора завершается.
		//if (_charToken.Type == CharType.EndOfStream)
		//{
		//	Type = Source.Symbol.Type.EndOfStream;
		//	return;
		//}

		//// В зависимости от текущего состояния необходимо определить следующую m-конфигурацию.
		//if (IsIdentifierLetter(false))
		//	ReadIdentifier();
		//else if (_charToken.IsWhitespace())
		//	ReadWhitespace();
		//else if (_charToken.IsEndOfLine())
		//	ReadEndOfLine();
		//else if (_charToken.IsDecimalDigit())
		//	ReadInteger(); // Наверное переделать в считывание числа в принципе любого для языка.
		//else
		//	ReadDelimiter();
	}
}

File: Proc-0/src/main/kotlin/Source/Proc/HelpSymbol.kt
package proc.Source.Proc

import proc.Source.SymbolReader
import proc.Text.CharStream

/**
 * Лексический анализатор документации.
 */
class HelpSymbol: SymbolReader {

	/**
	 * Основной конструктор.
	 * @param charToken Символьный поток.
	 */
	constructor(charToken: CharStream) : super(charToken) {
	}

	private fun ReadDelimiter() {
	}

	protected fun ReadNumber() {
	}

	/**
	 * Чтение очередной лексемы. Простейший шаг, определяющий следующий автомат.
	 * Результат чтения не имеет значения, т.к. синтаксический анализатор сам решает,
	 * является ли для него состояние машины приемлемым для следующего шага.
	 */
	public override fun Read() {
		//// Если нет возможности прочесть очередное состояние из потока, работа анализатора завершается.
		//if (_charToken.Type == CharType.EndOfStream)
		//{
		//	Type = Source.Symbol.Type.EndOfStream;
		//	return;
		//}

		//// В зависимости от текущего состояния необходимо определить следующую m-конфигурацию.
		//if (IsIdentifierLetter(false))
		//	ReadIdentifier();
		//else if (_charToken.IsWhitespace())
		//	ReadWhitespace();
		//else if (_charToken.IsEndOfLine())
		//	ReadEndOfLine();
		//else if (_charToken.IsDecimalDigit())
		//	ReadInteger(); // Наверное переделать в считывание числа в принципе любого для языка.
		//else
		//	ReadDelimiter();
	}
}
